cpp/oneapi/dal/backend/interop/table_conversion.hpp

```diff
@@ -122,12 +123,159 @@ inline daal::data_management::NumericTablePtr convert_to_daal_table(const homoge
+                                      array<std::int64_t>& row_indices,
+                                      std::int64_t row_count,
+                                      std::int64_t column_count,
+                                      bool allow_copy = false) {
+    if (!data.get_count() || !column_indices.get_count() || !row_indices.get_count())
+        return daal::services::SharedPtr<daal::data_management::CSRNumericTable>();
+
+    if (allow_copy) {
+        data.need_mutable_data();
+        column_indices.need_mutable_data();
+        row_indices.need_mutable_data();
+    }
+
+    ONEDAL_ASSERT(data.has_mutable_data());
+    ONEDAL_ASSERT(column_indices.has_mutable_data());
+    ONEDAL_ASSERT(row_indices.has_mutable_data());
+
+    const auto daal_data =
+        daal::services::SharedPtr<T>(data.get_mutable_data(), daal_object_owner{ data });
+    ONEDAL_ASSERT(sizeof(std::size_t) == sizeof(std::int64_t));
+    const auto daal_column_indices = daal::services::SharedPtr<std::size_t>(
+        reinterpret_cast<std::size_t*>(column_indices.get_mutable_data()),
+        daal_object_owner{ column_indices });
+    const auto daal_row_indices = daal::services::SharedPtr<std::size_t>(
+        reinterpret_cast<std::size_t*>(row_indices.get_mutable_data()),
+        daal_object_owner{ row_indices });
+
+    return daal::data_management::CSRNumericTable::create(
+        daal_data,
+        daal_column_indices,
+        daal_row_indices,
+        dal::detail::integral_cast<std::size_t>(column_count),
+        dal::detail::integral_cast<std::size_t>(row_count));
+}
+
+template <typename Float>
+inline daal::data_management::CSRNumericTablePtr copy_to_daal_csr_table(
+    const detail::csr_table& table) {
+    const bool allow_copy = true;
+    auto block = detail::csr_accessor<const Float>{ table }.pull();
+    return convert_to_daal_csr_table(block.data,
+                                     block.column_indices,
+                                     block.row_indices,
+                                     table.get_row_count(),
+                                     table.get_column_count(),
+                                     allow_copy);
+}
+
+template <typename T>
+inline detail::csr_table convert_from_daal_csr_table(
+    const daal::data_management::NumericTablePtr& nt) {
+    struct sparse_block_owner {
+        sparse_block_owner(const daal::data_management::NumericTablePtr& nt) {
+            _csr_nt = daal::services::dynamicPointerCast<daal::data_management::CSRNumericTable,
+                                                         daal::data_management::NumericTable>(nt);
+            ONEDAL_ASSERT(_csr_nt);
+            _row_count = _csr_nt->getNumberOfRows();
+            _column_count = _csr_nt->getNumberOfColumns();
+            _csr_nt->getSparseBlock(0, _row_count, daal::data_management::readOnly, _block);
+            _is_free = false;
+        }
+
+        void release() {
```

Comment:
Why you need this method? Why not use RAII practice here?

Reply:
It was fixed